lib/torrent.js

```diff
@@ -280,7 +281,15 @@ Torrent.prototype._onSwarmListening = function () {
     peerId: self.client.peerId,
     port: self.client.torrentPort,
     rtcConfig: self.client._rtcConfig,
-    wrtc: self.client._wrtc
+    wrtc: self.client._wrtc,
+    getAnnounceOpts: function () {
+      return {
+        uploaded: self.uploaded,
+        downloaded: self.downloaded,
+        left: self.length - self.downloaded,
+        extraAnnounceOpts: self._getExtraAnnounceOpts && self._getExtraAnnounceOpts()
```

Comment:
Instead of passing the user's custom announce opts separately in this `extraAnnounceOpts` property, can we merge their properties with the `uploaded`, `downloaded`, and `left` params?

Thinking something like this:

```js
getAnnounceOpts: function () {
  return extend({
    uploaded: self.uploaded,
    downloaded: self.downloaded,
    left: self.length - self.downloaded
  }, self._getExtraAnnounceOpts && self._getExtraAnnounceOpts())
})
```

This ordering will mean that the user's values for `uploaded`, `downloaded`, and `left` (if they exist) will take precedence.